src/Illuminate/Routing/Router.php

```diff
@@ -1201,6 +1201,23 @@ public function setRoutes(RouteCollection $routes)
         $this->container->instance('routes', $this->routes);
     }
 
+    /**
+     * Set the compiled route collection instance.
+     *
+     * @param  array  $routes
+     * @return void
+     */
+    public function setCompiledRoutes(array $routes)
+    {
+        foreach ($routes['attributes'] as ['methods' => $methods, 'uri' => $uri, 'action' => $action]) {
+            $this->routes->add($this->newRoute($methods, $uri, $action));
+        }
```

Comment:
You might notice that I'm re-resolving every route here instead of setting them like beforehand. I noticed that this doesn't really makes much of a difference if I did it like before so I might be missing something here.